### PR TITLE
SSCC label printing

### PR DESCRIPTION
**Pull request by metas-ts**
allow different GS1-manufactoring codes for different AD_Orgs
metasfresh/metasfresh#5896

### CODE REVIEW

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- de.metas.edi/src/main/java/de/metas/edi/api/ZebraPrinterSysConfigType.java  1
- de.metas.edi/src/main/java/de/metas/edi/api/ZebraPrinterService.java  1
- de.metas.edi/src/main/java/de/metas/edi/process/EDI_DesadvGenerateCSV_FileForSSCC_Labels.java  2
- de.metas.edi/src/main/java/de/metas/edi/process/EDI_GenerateCSV_FileForSSCC_Labels.java  2
- de.metas.edi/src/main/java/de/metas/edi/process/EDI_DesadvLinePackGenerateCSV_FileForSSCC_Labels.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metasfresh/metasfresh/pullRequest?prid=4648830&bid=15567600)

File: de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/util/DB.java [hunk 1]
```diff
@@ -3016,4 +3017,53 @@ public static boolean isDBColumnPresent(@NonNull final String tableName, @NonNul
 
 	}
 
+	/**
+	 * @param sqlStatement	SQL statement to be executed
+	 * @param parameters	Parameters to be used in the {@param sqlStatement}
+	 * @param trxName		transaction name
+	 * @return each resulted row as a {@link List<String>}
+	 */
+	public static List<List<String>> getSQL_ResultRowsAsListsOfStrings(final String sqlStatement, final List<Object> parameters, final String trxName)
```
Comment (metas-ts) [inline de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/util/DB.java]: use ImmutableList

File: de.metas.util/src/main/java/de/metas/util/StringUtils.java [hunk 2]
```diff
@@ -993,4 +993,12 @@ else if (capitalize)
 		}
 		return new String(data);
 	}	// initCap
+
+	/**
+	 * @param in	input {@link String}
+	 * @return {@param in} if != null, empty string otherwise
+	 */
+	public static String nonNullOrEmpty(final String in) {
```
Comment (metas-ts) [inline de.metas.util/src/main/java/de/metas/util/StringUtils.java]: we already have `de.metas.util.Check.isEmpty(String, boolean)` which can also handle whitespace strings. please use it instead.

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- de.metas.edi/src/main/java/de/metas/edi/process/EDI_DesadvLinePackGenerateCSV_FileForSSCC_Labels.java  2
- de.metas.edi/src/main/java/de/metas/edi/process/EDI_DesadvGenerateCSV_FileForSSCC_Labels.java  2
- de.metas.edi/src/main/java/de/metas/edi/process/EDI_GenerateCSV_FileForSSCC_Labels.java  2
- de.metas.edi/src/main/java/de/metas/edi/api/ZebraPrinterSysConfigType.java  1
- de.metas.edi/src/main/java/de/metas/edi/api/ZebraPrinterService.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metasfresh/metasfresh/pullRequest?prid=4648830&bid=15567600)

File: de.metas.util/src/main/java/de/metas/util/StringUtils.java (hunk 2 above)
Comment (pvpurcarcosmin) [reply de.metas.util/src/main/java/de/metas/util/StringUtils.java]: They are not doing the same thing; I need an method that will return an empty String if the input String is null

Comment (pvpurcarcosmin) [reply de.metas.util/src/main/java/de/metas/util/StringUtils.java]: I think I should find a better name for the method since this one seems misleading.

**Comment by metas-ts** [comment]
the not-yet-build build contains no SQL that might fail => integrating now

**Pull request merged**